src/range.rs

```diff
@@ -223,142 +247,62 @@ fn bound_as_ref<V>(bound: &Bound<V>) -> Bound<&V> {
     }
 }
 
+fn valid_segment<T: PartialOrd>(start: &Bound<T>, end: &Bound<T>) -> bool {
+    match (start, end) {
+        (Included(s), Included(e)) => s <= e,
+        (Included(s), Excluded(e)) => s < e,
+        (Excluded(s), Included(e)) => s < e,
+        (Excluded(s), Excluded(e)) => s < e,
+        (Unbounded, _) | (_, Unbounded) => true,
```

Comment:
Actually, there might be representations special cases where we have an invalid segment that cannot be checked by this function like `(Unbounded, Excluded(0))` in the case where versions are `u32`. It depends if there are other ways to represent the left-most and right-most bounds with the version type.

What do you guys think of that? negligible?